localstack/aws/connect.py

```diff
@@ -0,0 +1,473 @@
+            aws_secret_access_key=aws_secret_access_key or INTERNAL_AWS_SECRET_ACCESS_KEY,
+            aws_session_token=aws_session_token,
+            config=config or self._config,
+        )
+
+
+class ExternalClientFactory(ClientFactory):
+    def get_client(
+        self,
+        service_name: str,
+        region_name: Optional[str],
+        aws_access_key_id: Optional[str] = None,
+        aws_secret_access_key: Optional[str] = None,
+        aws_session_token: Optional[str] = None,
+        endpoint_url: str = None,
+        config: Config = None,
+    ) -> BaseClient:
+        """
+        Build and return client for connections originating outside LocalStack.
+
+        If either of the access keys or region are set to None, they are loaded from following
+        locations:
+        - AWS environment variables
+        - Credentials file `~/.aws/credentials`
+        - Config file `~/.aws/config`
+
+        :param service_name: Service to build the client for, eg. `s3`
+        :param region_name: Name of the AWS region to be associated with the client
+            If set to None, loads from botocore session.
+        :param aws_access_key_id: Access key to use for the client.
+            If set to None, loads from botocore session.
+        :param aws_secret_access_key: Secret key to use for the client.
+            If set to None, loads from botocore session.
+        :param aws_session_token: Session token to use for the client.
+            Not being used if not set.
+        :param endpoint_url: Full endpoint URL to be used by the client.
+            Defaults to appropriate LocalStack endpoint.
+        :param config: Boto config for advanced use.
+        """
+
+        return self._get_client(
+            service_name=service_name,
+            region_name=region_name or self._get_region(),
+            use_ssl=self._use_ssl,
+            verify=self._verify,
+            endpoint_url=endpoint_url or get_local_service_url(service_name),
+            aws_access_key_id=aws_access_key_id,
+            aws_secret_access_key=aws_secret_access_key,
+            aws_session_token=aws_session_token,
+            config=config or self._config,
+        )
+
+
+connect_to = InternalClientFactory()
+connect_externally_to = ExternalClientFactory()
+
+#
+# Handlers
+#
+
+
+def _handler_create_request_parameters(params, model, context, **kwargs):
+    """
+    Construct the data transfer object at the time of parsing the client
+    parameters and proxy it via the Boto context dict.
+
+    This handler enables the use of additional keyword parameters in Boto API
+    operation functions.
+    """
+
+    # Names of arguments that can be passed to Boto API operation functions.
+    # These must correspond to entries on the data transfer object.
+    dto = InternalRequestParameters()
+    for member in InternalRequestParameters.__annotations__.keys():
+        parameter = f"_{''.join([part.title() for part in member.split('_')])}"
```

Comment:
Could you maybe add a short doc about this line to the docstring of the function?
If I read it correctly, this means that you can set the individual internal request parameters by using an `_` prefix, right?

Reply:
You are right! A parameter `_SourceArn` will be serialized in the dto as `source_arn`